jetty-server/src/main/java/org/eclipse/jetty/server/handler/FileBufferedResponseHandler.java

```diff
@@ -106,18 +109,9 @@ protected void dispose()
             _fileOutputStream = null;
             _aggregating = null;
 
-            if (_filePath != null)
-            {
-                try
-                {
-                    Files.delete(_filePath);
-                }
-                catch (Throwable t)
-                {
-                    LOG.warn("Could not delete file {}", _filePath, t);
-                }
-                _filePath = null;
-            }
+            ReferenceCleaner.DeleteFile.delete(_filePath);
```

Comment:
Maybe we should only register the file with the Cleaner if this delete fails. This way we should not have anything registered with the Cleaner except on windows.

Reply:
I tried that, the .reset() and .resetBuffer() use cases made that approach difficult.
Besides, the line you flagged just deletes the file directly, no register step.

Reply:
> Besides, the line you flagged just deletes the file directly, no register step.

I meant to only register here if this line fails to delete the file, and to not register it with the Cleaner when the file is created.


> I tried that, the .reset() and .resetBuffer() use cases made that approach difficult.

What do you mean by the .reset() and .resetBuffer() cases?